src/Adapter/Image/Uploader/SupplierImageUploader.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Image\Uploader;
+
+use Configuration;
+use Context;
+use ImageManager;
+use ImageType;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\ImageOptimizationException;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\ImageUploadException;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\MemoryLimitException;
+use PrestaShopException;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+/**
+ * Uploads supplier logo image
+ */
+final class SupplierImageUploader extends AbstractImageUploader
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function upload($supplierId, UploadedFile $image)
+    {
+        $this->checkImageIsAllowedForUpload($image);
+        $temporaryImageName = tempnam(_PS_TMP_IMG_DIR_, 'PS');
```

Comment:
@PierreRambaud Is it OK with security best practices ? Looks like an eligible LFI attack vector

(now that I know this name I'm going to put it everywhere 😄)

Reply:
Nope, you can do it, tempnam generate a random name depending in the destination / prefix.
And for because move_uploaded_file move the file to the temporary location
it's like mktemp for linux.